### DISCUSSION TITLE
How to keep track of reconnected UEs

### DISCUSSION THREAD

**Discussion by HaoxinSEU** [srsran/srsRAN_Project#577] [General Help]
Hi community,

We would like to keep track of UEs in the gNB. From our understanding, many unique variables like `rnti`, `du_ue_index` can represent one specific UE. 

However, if the UE reconnects, all these variables are re-generated and new, so that we can not link these new values to the old UE. In other words, if we use these values, the reconnection will be considered as a new UE, instead of a previous UE. 

So to track one UE even with reconnection, there should be a state that doesn't change during reconnection. We thought of `imsi` before, but they are not recorded in the gNB.

Does anyone have an idea about which variable/state we can use for this application? Thanks for your help!

**Comment by pgawlowicz** [comment]
Hi, 
If the UE reconnects quickly, it sends C-RNTI in the RRC Reestablishment Request to recover the old UE context at gNB.
